src/features/device-proxy/device-proxy.ts

```diff
@@ -146,12 +155,30 @@ interface FixedMethodRequestDevicesOpts {
 	data?: AnyObject;
 	req?: sbvrUtils.Passthrough['req'];
 	wait?: boolean;
+	appId?: number;
 }
 
 interface RequestDevicesOpts extends FixedMethodRequestDevicesOpts {
 	method: string;
 }
 
+// Check user resource access via /canAccess
+async function checkResourceAccess(
+	client: typeof api.resin,
+	resource: 'application' | 'device',
+	id: number,
+): Promise<void> {
+	const res = (await client.request({
+		method: 'POST',
+		url: `${resource}(${id})/canAccess`,
```

Comment:
@otaviojacobi isn't there a nice way of doing `canAccess` calls via pinejs-client now? Or was that only specific to the permissions?

Reply:
Afaict it's only supported inside `$filter` (since the implementation was added in `handleFilterOperator`).
See: https://github.com/balena-io-modules/pinejs-client-js/pull/183/files